redash/query_runner/qubole.py

```diff
@@ -59,16 +68,26 @@ def annotate_query(cls):
 
     def test_connection(self):
         headers = self._get_header()
-        r = requests.head("%s/api/latest/users" % self.configuration['endpoint'], headers=headers)
+        r = requests.head("%s/api/latest/users" % self.configuration.get('endpoint'), headers=headers)
         r.status_code == 200
 
     def run_query(self, query, user):
-        qbol.configure(api_token=self.configuration['token'],
-                       api_url='%s/api' % self.configuration['endpoint'])
+        qbol.configure(api_token=self.configuration.get('token'),
+                       api_url='%s/api' % self.configuration.get('endpoint'))
 
         try:
-            cls = PrestoCommand if(self.configuration['query_type'] == 'presto') else HiveCommand
-            cmd = cls.create(query=query, label=self.configuration['cluster'])
+            query_type = self.configuration.get('query_type')
```

Comment:
```suggestion
            query_type = self.configuration.get('query_type', 'hive')
```

You need to provide a default value. Using `hive` here to be consistent with previous version and not break existing deployments (see previous comment).

Reply:
Done